src/service/viewport/viewport-impl.js

```diff
@@ -501,43 +513,134 @@ export class Viewport {
+   * @return {!Promise}
+   * @private
+   */
+  animateScrollIntoViewInternal_(element, parent, duration, curve, pos) {
     const elementRect = this.binding_.getLayoutRect(element);
+
+    const {height: parentHeight} = this.isScrollingElement_(parent) ?
+      this.getSize() :
+      this.getLayoutRect(parent);
+
     let offset;
     switch (pos) {
       case 'bottom':
-        offset = -this.getHeight() + elementRect.height;
+        offset = -parentHeight + elementRect.height;
         break;
       case 'center':
-        offset = (-this.getHeight() / 2) + (elementRect.height / 2);
+        offset = (-parentHeight / 2) + (elementRect.height / 2);
         break;
       default:
         offset = 0;
         break;
     }
     let newScrollTop;
-    let curScrollTop;
+    let curScrollTopPromise;
 
     if (this.useLayers_) {
       newScrollTop = elementRect.top + offset;
-      curScrollTop = 0;
+      curScrollTopPromise = Promise.resolve(0);
     } else {
       const calculatedScrollTop = elementRect.top - this.paddingTop_ + offset;
       newScrollTop = Math.max(0, calculatedScrollTop);
-      curScrollTop = this.getScrollTop();
-    }
-    if (newScrollTop == curScrollTop) {
-      return Promise.resolve();
+      curScrollTopPromise = this.getElementScrollTop_(parent);
     }
+
+    return curScrollTopPromise.then(curScrollTop => {
+      if (newScrollTop == curScrollTop) {
+        return;
+      }
+      return this.interpolateScrollIntoView_(
+          parent, curScrollTop, newScrollTop, duration, curve);
+    });
+  }
+
+  /**
+   * @param {!Element} parent
+   * @param {number} curScrollTop
+   * @param {number} newScrollTop
+   * @param {number} duration
+   * @param {string} curve
+   * @private
+   */
+  interpolateScrollIntoView_(
+    parent, curScrollTop, newScrollTop, duration, curve) {
+
     /** @const {!TransitionDef<number>} */
     const interpolate = numeric(curScrollTop, newScrollTop);
-    // TODO(aghassemi, #10463): the duration should not be a constant and should
-    // be proportional to the distance to be scrolled for better transition
-    // experience when things are closer vs farther.
-    return Animation.animate(this.ampdoc.getRootNode(), position => {
-      this.binding_.setScrollTop(interpolate(position));
+
+    // TODO(aghassemi, #10463): the duration should not be a constant and
+    // should be proportional to the distance to be scrolled for better
+    // transition experience when things are closer vs farther.
+    return Animation.animate(parent, position => {
+      this.setElementScrollTop_(parent, interpolate(position));
     }, duration, curve).then();
   }
 
+  /**
+   * @param {!Element} element
+   * @return {!Promise<!Element>}
+   */
+  getScrollingContainerFor_(element) {
+    return this.vsync_.measurePromise(() => {
+      const scrollable = closest(element,
+          parent => parent./*OK*/scrollHeight > parent./*OK*/clientHeight,
```

Comment:
this will be true if `overflow` is `visible` (default) or `hidden` as well. I think we need to check computed style for `overflow-y` and make sure it is either `auto` or `scroll`.

https://jsbin.com/pelakumoji/edit?html,css,js,console,output

Reply:
Oh and just thought of another issue: if the parent itself is not in view, parent won't get scrolled to in the first place. Honestly it will be cheaper and safer to special case sidebar and lighbox[scrollable] than trying to solve this in a generic way.

Reply:
Special cased, ptal.